R/read_redcap.R

```diff
@@ -440,18 +443,21 @@ add_metadata <- function(supertbl, db_metadata, redcap_uri, token, suppress_redc
 #' @param supertbl a supertibble object to supplement with metadata
 #' @param linked_arms the tibble with event mappings created by
 #' \code{link_arms()}
+#' @param repeat_event_types a dataframe output from [repeat_event_types()] which
+#' specifies NR, RS, and RT types for events
 #'
 #' @return
 #' The original supertibble with an events \code{redcap_events} list column
 #' containing arms and events associated with each instrument
 #'
 #' @keywords internal
-#'
-add_event_mapping <- function(supertbl, linked_arms) {
+
+add_event_mapping <- function(supertbl, linked_arms, repeat_event_types) {
```

Comment:
`add_event_mapping()` now accepts a reference table for the repeating event types designations. This table is joined onto the `linked_arms` data to make `repeat_type` available in the eventual `redcap_events` supertibble column.